File: 0260-single-number-iii/0260-single-number-iii.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        vector<int> selected;
        sort(nums.begin(), nums.end());
        for(int i = 0; i < nums.size(); i+=2) {
            if(i == nums.size() - 1 || nums[i] < nums[i + 1]) {
                selected.push_back(nums[i--]);
            }
            if(selected.size() == 2) break;
        }
        
        return selected;
    }
};

File: 0260-single-number-iii/0260-single-number-iii.java
class Solution {
    public int[] singleNumber(int[] nums) {
        Arrays.sort(nums);
        int[] selected = new int[2];
        int index = 0;
        for(int i = 0; i < nums.length; i+=2) {
            if(i == nums.length - 1 || nums[i] != nums[i + 1]) {
                selected[index++] = nums[i--];
            }
            if(index == 2) break;
        }
        
        return selected;
    }
}